docker-entrypoint.sh

```diff
@@ -11,5 +11,6 @@ if [ "$1" = 've/bin/gunicorn' ] || [ "$1" = 've/bin/python' ]; then
   ve/bin/python manage.py migrate notifications
   set -e
   ve/bin/python manage.py migrate workbench
+  (cd specifyweb && ../ve/bin/python ../manage.py compilemessages)
```

Comment:
Is there some reason this step can't be done in the build-backend stage of the Dockerfile?

Reply:
I was getting an error when doing it there. I don't remember the error message. Would try to replicate and post it here

Either way, message compilation should not take much time

Reply:
Here is the exception:
```yaml
~/site/python/specify7/specifyweb/frontend/js_src/lib backend-localization* ❯ dcu --build
[2] 56368
appending output to nohup.out
[+] Building 3.7s (47/83)
 => [specify7_webpack internal] load build definition from Dockerfile                                             0.0s
 => => transferring dockerfile: 32B                                                                               0.0s
 => [specify7_specify7-worker internal] load build definition from Dockerfile                                     0.0s
 => => transferring dockerfile: 32B                                                                               0.0s
 => [specify7_specify7 internal] load build definition from Dockerfile                                            0.0s
 => => transferring dockerfile: 32B                                                                               0.0s
 => [specify7_webpack internal] load .dockerignore                                                                0.0s
 => => transferring context: 35B                                                                                  0.0s
 => [specify7_specify7-worker internal] load .dockerignore                                                        0.0s
 => => transferring context: 35B                                                                                  0.0s
 => [specify7_specify7 internal] load .dockerignore                                                               0.0s
 => => transferring context: 35B                                                                                  0.0s
 => [specify7_webpack internal] load metadata for docker.io/library/node:16.1.0-alpine3.11                        1.8s
 => [specify7_specify7-worker internal] load metadata for docker.io/library/ubuntu:18.04                          1.7s
 => [auth] library/node:pull token for registry-1.docker.io                                                       0.0s
 => [auth] library/ubuntu:pull token for registry-1.docker.io                                                     0.0s
 => [specify7_specify7-worker build-frontend 1/7] FROM docker.io/library/node:16.1.0-alpine3.11@sha256:0b1a98a47  0.0s
 => [specify7_webpack internal] load build context                                                                0.2s
 => => transferring context: 45.83kB                                                                              0.1s
 => [specify7_specify7-worker common 1/5] FROM docker.io/library/ubuntu:18.04@sha256:139b3846cee2e63de9ced83cee7  0.0s
 => [specify7_specify7 internal] load build context                                                               0.2s
 => => transferring context: 66.33kB                                                                              0.1s
 => [specify7_specify7-worker internal] load build context                                                        0.2s
 => => transferring context: 66.33kB                                                                              0.1s
 => CACHED [specify7_specify7-worker build-frontend 2/7] WORKDIR /home/node                                       0.0s
 => CACHED [specify7_webpack build-frontend 3/7] COPY --chown=node:node specifyweb/frontend/js_src/package.json   0.0s
 => CACHED [specify7_webpack build-frontend 4/7] RUN npm install                                                  0.0s
 => CACHED [specify7_webpack build-frontend 5/7] RUN mkdir dist && chown node:node dist                           0.0s
 => CACHED [specify7_webpack build-frontend 6/7] COPY --chown=node:node specifyweb/frontend/js_src .              0.0s
 => CACHED [specify7_webpack build-frontend 7/7] RUN npx webpack                                                  0.0s
 => [specify7_webpack] exporting to image                                                                         0.0s
 => => exporting layers                                                                                           0.0s
 => => writing image sha256:8620b090df652e913734a8e4c5189178e2846023120fefb70f2d9b7710aa88a6                      0.0s
 => => naming to docker.io/library/specify7_webpack                                                               0.0s
 => CACHED [specify7_specify7-worker common 2/5] RUN apt-get update  && apt-get -y install --no-install-recommen  0.0s
 => CACHED [specify7_specify7-worker common 3/5] RUN groupadd -g 999 specify  && useradd -r -u 999 -g specify sp  0.0s
 => CACHED [specify7_specify7-worker common 4/5] RUN mkdir -p /home/specify  && chown specify.specify /home/spec  0.0s
 => CACHED [specify7_specify7-worker common 5/5] RUN mkdir -p /opt/specify7  && chown specify.specify /opt/speci  0.0s
 => CACHED [specify7_specify7-worker build-backend  1/16] RUN apt-get update  && apt-get -y install --no-install  0.0s
 => CACHED [specify7_specify7-worker build-backend  2/16] COPY --chown=specify:specify requirements.txt /home/sp  0.0s
 => CACHED [specify7_specify7-worker build-backend  3/16] WORKDIR /opt/specify7                                   0.0s
 => CACHED [specify7_specify7-worker build-backend  4/16] RUN python3.6 -m venv ve  && ve/bin/pip install --no-c  0.0s
 => CACHED [specify7_specify7-worker build-backend  5/16] RUN ve/bin/pip install --no-cache-dir gunicorn redis    0.0s
 => CACHED [specify7_specify7-worker build-frontend 3/7] COPY --chown=node:node specifyweb/frontend/js_src/packa  0.0s
 => CACHED [specify7_specify7-worker build-frontend 4/7] RUN npm install                                          0.0s
 => CACHED [specify7_specify7-worker build-frontend 5/7] RUN mkdir dist && chown node:node dist                   0.0s
 => CACHED [specify7_specify7-worker build-frontend 6/7] COPY --chown=node:node specifyweb/frontend/js_src .      0.0s
 => CACHED [specify7_specify7-worker build-frontend 7/7] RUN npx webpack                                          0.0s
 => CACHED [specify7_specify7-worker build-backend  6/16] COPY --from=build-frontend /home/node/dist specifyweb/  0.0s
 => CACHED [specify7_specify7-worker build-backend  7/16] COPY --chown=specify:specify specifyweb /opt/specify7/  0.0s
 => CACHED [specify7_specify7-worker build-backend  8/16] COPY --chown=specify:specify manage.py /opt/specify7/   0.0s
 => CACHED [specify7_specify7-worker build-backend  9/16] COPY --chown=specify:specify docker-entrypoint.sh /opt  0.0s
 => CACHED [specify7_specify7-worker build-backend 10/16] COPY --chown=specify:specify Makefile /opt/specify7/    0.0s
 => CACHED [specify7_specify7-worker build-backend 11/16] COPY --chown=specify:specify specifyweb.wsgi /opt/spec  0.0s
 => CACHED [specify7_specify7-worker build-backend 12/16] RUN make specifyweb/settings/build_version.py           0.0s
 => CACHED [specify7_specify7-worker build-backend 13/16] RUN echo  > specifyweb/frontend/static/build_version.t  0.0s
 => CACHED [specify7_specify7-worker build-backend 14/16] RUN echo $GIT_SHA > specifyweb/frontend/static/git_sha  0.0s
 => CACHED [specify7_specify7-worker build-backend 15/16] RUN date > specifyweb/frontend/static/build_date.txt    0.0s
 => ERROR [specify7_specify7-worker build-backend 16/16] RUN (cd specifyweb && ../ve/bin/python ../manage.py com  1.2s
------
 > [specify7_specify7-worker build-backend 16/16] RUN (cd specifyweb && ../ve/bin/python ../manage.py compilemessages):
#75 1.025 Traceback (most recent call last):
#75 1.025   File "/opt/specify7/ve/lib/python3.6/site-packages/django/core/management/base.py", line 323, in run_from_argv
#75 1.025     self.execute(*args, **cmd_options)
#75 1.025   File "/opt/specify7/ve/lib/python3.6/site-packages/django/core/management/base.py", line 364, in execute
#75 1.025     output = self.handle(*args, **options)
#75 1.025   File "/opt/specify7/ve/lib/python3.6/site-packages/django/core/management/commands/compilemessages.py", line 64, in handle
#75 1.025     basedirs.extend(settings.LOCALE_PATHS)
#75 1.025   File "/opt/specify7/ve/lib/python3.6/site-packages/django/conf/__init__.py", line 79, in __getattr__
#75 1.025     self._setup(name)
#75 1.025   File "/opt/specify7/ve/lib/python3.6/site-packages/django/conf/__init__.py", line 66, in _setup
#75 1.025     self._wrapped = Settings(settings_module)
#75 1.025   File "/opt/specify7/ve/lib/python3.6/site-packages/django/conf/__init__.py", line 157, in __init__
#75 1.025     mod = importlib.import_module(self.SETTINGS_MODULE)
#75 1.025   File "/usr/lib/python3.6/importlib/__init__.py", line 126, in import_module
#75 1.025     return _bootstrap._gcd_import(name[level:], package, level)
#75 1.025   File "<frozen importlib._bootstrap>", line 994, in _gcd_import
#75 1.025   File "<frozen importlib._bootstrap>", line 971, in _find_and_load
#75 1.025   File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
#75 1.025   File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
#75 1.025   File "<frozen importlib._bootstrap_external>", line 678, in exec_module
#75 1.025   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
#75 1.025   File "/opt/specify7/specifyweb/settings/__init__.py", line 32, in <module>
#75 1.025     from .secret_key import SECRET_KEY
#75 1.025 ModuleNotFoundError: No module named 'specifyweb.settings.secret_key'
#75 1.025
#75 1.025 During handling of the above exception, another exception occurred:
#75 1.025
#75 1.025 Traceback (most recent call last):
#75 1.025   File "../manage.py", line 22, in <module>
#75 1.025     execute_from_command_line(sys.argv)
#75 1.025   File "/opt/specify7/ve/lib/python3.6/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
#75 1.025     utility.execute()
#75 1.025   File "/opt/specify7/ve/lib/python3.6/site-packages/django/core/management/__init__.py", line 375, in execute
#75 1.025     self.fetch_command(subcommand).run_from_argv(self.argv)
#75 1.025   File "/opt/specify7/ve/lib/python3.6/site-packages/django/core/management/base.py", line 336, in run_from_argv
#75 1.025     connections.close_all()
#75 1.025   File "/opt/specify7/ve/lib/python3.6/site-packages/django/db/utils.py", line 219, in close_all
#75 1.025     for alias in self:
#75 1.025   File "/opt/specify7/ve/lib/python3.6/site-packages/django/db/utils.py", line 213, in __iter__
#75 1.025     return iter(self.databases)
#75 1.025   File "/opt/specify7/ve/lib/python3.6/site-packages/django/utils/functional.py", line 80, in __get__
#75 1.025     res = instance.__dict__[self.name] = self.func(instance)
#75 1.025   File "/opt/specify7/ve/lib/python3.6/site-packages/django/db/utils.py", line 147, in databases
#75 1.025     self._databases = settings.DATABASES
#75 1.025   File "/opt/specify7/ve/lib/python3.6/site-packages/django/conf/__init__.py", line 79, in __getattr__
#75 1.025     self._setup(name)
#75 1.025   File "/opt/specify7/ve/lib/python3.6/site-packages/django/conf/__init__.py", line 66, in _setup
#75 1.025     self._wrapped = Settings(settings_module)
#75 1.025   File "/opt/specify7/ve/lib/python3.6/site-packages/django/conf/__init__.py", line 157, in __init__
#75 1.025     mod = importlib.import_module(self.SETTINGS_MODULE)
#75 1.025   File "/usr/lib/python3.6/importlib/__init__.py", line 126, in import_module
#75 1.025     return _bootstrap._gcd_import(name[level:], package, level)
#75 1.025   File "<frozen importlib._bootstrap>", line 994, in _gcd_import
#75 1.025   File "<frozen importlib._bootstrap>", line 971, in _find_and_load
#75 1.025   File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
#75 1.025   File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
#75 1.025   File "<frozen importlib._bootstrap_external>", line 678, in exec_module
#75 1.025   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
#75 1.025   File "/opt/specify7/specifyweb/settings/__init__.py", line 32, in <module>
#75 1.025     from .secret_key import SECRET_KEY
#75 1.025 ModuleNotFoundError: No module named 'specifyweb.settings.secret_key'
------
failed to solve: rpc error: code = Unknown desc = executor failed running [/bin/sh -c (cd specifyweb && ../ve/bin/python ../manage.py compilemessages)]: exit code: 1
~/site/python/specify7/specifyweb/frontend/js_src/lib

```